### PR TITLE
Add a `getHistoricalEvents` API (port Swift changes)

### PR DESCRIPTION
**Pull request by yangyansong-adbe**
Exposes a `getHistoricalEvents` API in ExtensionApi class to follow the Swift implementation.
## Related Issue
-  #150

### CODE REVIEW

**Review by praveek — CHANGES_REQUESTED** [review]
Can you also update the historical condition to query `from extensionAPI.getHistoricalEvents` API?

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/EventHub.kt [hunk 1]
```diff
@@ -182,7 +198,15 @@ internal class EventHub {
             )
         }
 
-        // TODO: Record event to event history database if required.
+        event.mask?.let {
```
Comment (praveek) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/eventhub/EventHub.kt:201]: This should be moved to SerialWorkDispatcher.WorkHandler. We already have a Todo marked there for this logic. 
By keeping the logic here, we will miss recording events generated as result of dispatch consequence.

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/json/JSONCondition.kt [hunk 2]
```diff
@@ -38,30 +39,34 @@ internal abstract class JSONCondition {
...
+                            extensionApi
                         )
                     )
                     TYPE_VALUE_MATCHER -> MatcherCondition(
                         JSONDefinition.buildDefinitionFromJSON(
                             jsonCondition.getJSONObject(
                                 KEY_DEFINITION
-                            )
+                            ),
+                            extensionApi
```
Comment (spoorthipujariadobe) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/json/JSONCondition.kt:59]: Has `MatcherCondition` and `GroupCondition` been changed to accept `extensionApi`? Don't see it in this PR

**Review by spoorthipujariadobe — APPROVED** [review]

**Review by praveek — APPROVED** [review]

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/json/JSONCondition.kt (hunk 2 above)
Comment (yangyansong-adbe) [reply code/android-core-library/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/json/JSONCondition.kt:59]: No, we don't need to add it to other conditions as they don't depend on `ExtensionApi`.

**Pull request merged**